scipy/stats/tests/common_tests.py

```diff
@@ -199,6 +199,15 @@ def check_random_state_property(distfn, args):
     r2 = distfn.rvs(*args, size=8)
     npt.assert_equal(r0, r2)
 
+    # check that np.random.Generator can be used (numpy >= 1.17)
```

Comment:
wouldn't we want this to be an xfail type of situation that we would mark based on numpy version? Maybe I'm misunderstanding the intent of the code but it doesn't look like it could ever fail, am I missing something?

Reply:
`np.random.Generator` only came in 1.17, so the test shouldn't be run if numpy is too old, but we want it to run if it's available. I don't think marking the whole test as xfail (for numpy version) is right here, because it would make the whole test redundant on older numpy. So this specific test would have to be moved into its own function and marked as xfail. However, I don't think that can be done either because `check_random_state_property` is not a  pytest, it's only used by a parent pytest.

Reply:
ah, I think I understand now. The rng streams differ so we cannot have a test here that the values are equal or evan an almost equal. Is that right? 
I like the `hasattr()` approach, between that and the comment in line above about numpy version it will be more clear what is happening. 
I'm still stuck on an assertion here that could fail, but we'd want it only to fail on environments w/newer numpy.
The scope of this PR it seems is really upgrading numpy random api inside SciPy and not tests per se. 
I also don't have any idea here and I don't want to hold up a perfectly good improvement on this issue if I can't offer a solution. Hopefully someone can come up with a way to do the test in a future PR. So let's call this comment non-blocking.